src/VisualBoard.ts

```diff
@@ -1,18 +1,36 @@
 
-export function render(board: Array<Array<number>>) {
+export function buildGraphicalBoard(board: Array<Array<number>>): String {
     const alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
-    const lettresEntete = ' ' + alphabet.substr(0, board[0].length)
-    const entete = lettresEntete.split("").map(char => ` ${char} `)
-    const firstLine = entete.join("")
-
-    console.log(firstLine)
+    const headerLetter = ' ' + alphabet.substr(0, board[0].length)
+    const header = headerLetter.split("").map(char => ` ${char} `)
+    const firstLine = header.join("")
+    
+    let result = firstLine + '\n';
 
     for (let i = 0; i < board.length; i++) {
         let line = ` ${i} `;
-        for (const value of board[i]) {
-            line += ` ${value} `
+        for (const marble of board[i]) {
+            switch (marble) {
```

Comment:
you should definitely extract this to another function, like "marbleValuetoANSIColorCode". this would make the code clearer without the need for comments.